orangecontrib/prototypes/widgets/owcreatetable.py

```diff
@@ -160,21 +220,44 @@ class Outputs:
         data = Output("Data", Table)
 
     class Error(OWWidget.Error):
-        transform_err = Msg("Cannot transform input data into table!")
+        transform_err = Msg("Data does not fit to domain")
+
+    settingsHandler = PerfectDomainContextHandler()
 
-    n_rows = Setting(1)
-    n_columns = Setting(1)
+    n_rows = Setting(len(DEFAULT_DATA))
+    n_columns = Setting(len(DEFAULT_DATA[0]))
     auto_commit = Setting(True)
+    # since data is a small (at most 20x20) table we can afford to store it
```

Comment:
When input domain is larger than 20 variables, that would not be the case. The number of columns is arbitrarily large. The number of rows, however, is fixed to 20. Is that still a limitation is this case?

Reply:
When the input domain is larger we allow more columns. I think it is OK, nobody would have a patience to fill 100 columns and 20 rowns manually.